### PR TITLE
feat: export all features in project

### PR DESCRIPTION
**Pull request by kwasniew**
## About the changes

Project export allows to export all project features regardless of what's selected. Use bulk export for selection-based export.
Screenshot 2023-12-18 at 14 39 03

Dialog shows that you're exporting all project feature toggles.
Screenshot 2023-12-18 at 14 39 08

https://github.com/Unleash/unleash/assets/1394682/1ab6a0aa-161a-4492-8da8-871b7781c83a

### CODE REVIEW

File: src/lib/features/export-import-toggles/export-import-service.ts [hunk 1]
```diff
@@ -786,10 +786,21 @@ export default class ExportImportService
         userName: string,
         userId: number,
     ): Promise<ExportResultSchema> {
-        const featureNames =
-            typeof query.tag === 'string'
-                ? await this.featureTagService.listFeatures(query.tag)
-                : (query.features as string[]) || [];
+        let featureNames: string[] = [];
+        if (typeof query.tag === 'string') {
```
Comment (kwasniew) [inline src/lib/features/export-import-toggles/export-import-service.ts:790]: tag has a highest precedence

File: src/lib/features/export-import-toggles/export-import-service.ts [hunk 2]
```diff
@@ -786,10 +786,21 @@ export default class ExportImportService
...
+            featureNames = await this.featureTagService.listFeatures(query.tag);
+        } else if (Array.isArray(query.features) && query.features.length) {
```
Comment (kwasniew) [inline src/lib/features/export-import-toggles/export-import-service.ts:792]: then we look for explicit feature names (non empty list)

File: src/lib/features/export-import-toggles/export-import-service.ts [hunk 3]
```diff
@@ -786,10 +786,21 @@ export default class ExportImportService
...
+            featureNames = query.features;
+        } else if (typeof query.project === 'string') {
+            const allProjectFeatures = await this.toggleStore.getAll({
```
Comment (kwasniew) [inline src/lib/features/export-import-toggles/export-import-service.ts:795]: new capability to search by all project features

File: src/lib/features/export-import-toggles/export-import-service.ts [hunk 4]
```diff
@@ -786,10 +786,21 @@ export default class ExportImportService
...
+                project: query.project,
+            });
+            featureNames = allProjectFeatures.map((feature) => feature.name);
+        } else {
+            const allFeatures = await this.toggleStore.getAll();
```
Comment (kwasniew) [inline src/lib/features/export-import-toggles/export-import-service.ts:800]: fallback to all features

**Review by sjaanus — APPROVED** [review]

**Pull request merged**